mava/components/jax/training/step.py

```diff
@@ -56,6 +65,21 @@ def name() -> str:
         """Static method that returns component name."""
         return "step"
 
+    @staticmethod
+    def required_components() -> List[Type[Callback]]:
+        """List of other Components required in the system for this Component to function.
+
+        TrajectoryDataset required to set up trainer.store.dataset_iterator.
+        Step required to set up trainer.store.step_fn.
+        TrainerParameterClient required to set up trainer.store.trainer_parameter_client
+        and trainer.store.trainer_counts.
+        Logger required to set up trainer.store.trainer_logger.
+
+        Returns:
+            List of required component classes.
+        """
+        return [TrajectoryDataset, Step, TrainerParameterClient, Logger]
+
 
 class DefaultTrainerStep(TrainerStep):
```

Comment:
Just a general comment: there are too many step components in this class and made more confusing since TrainerStep is not an extension of Step and has a dependency on it

Reply:
Might be good to create an issue to rename some of the `Step` classes? `TrainerStep ` vs `Step` are not great names for disambiguation, I agree.

Reply:
Yeah that would be important

Reply:
@mmorris44 https://github.com/instadeepai/Mava/issues/678